### PR TITLE
GH Actions: restrict access to certain GH Actions

### PR DESCRIPTION
**Pull request by yoomlam**
## What was the problem?

Anyone on the current team are allowed to do potentially dangerous actions, such as deploy to prod, via GH Action workflows. 

Associated tickets or Slack threads:
- #1961

## How does this fix it?

Only allow members of the [VRO-RESTRICTED](https://github.com/orgs/department-of-veterans-affairs/teams/vro-restricted/members) team to do potentially dangerous actions.

Affected workflows:
- `delete-old-releases.yml`
- `delete-published-images.yml`
- `delete-workflow-runs.yml`
- `deploy-secrets.yml` - restricted only when deploying to `prod`
- `update-deployment.yml` - restricted only when deploying to `prod`

## How to test this PR
If you are not in [VRO-RESTRICTED](https://github.com/orgs/department-of-veterans-affairs/teams/vro-restricted/members), then the above GH Action workflows should fail (when run on this PR's branch `yoom/restrict-access`) with a `Only VRO-RESTRICTED team members can run this action` message.

### CODE REVIEW

**Comment by yoomlam** [comment]
These succeed b/c I'm in the [VRO-RESTRICTED](https://github.com/orgs/department-of-veterans-affairs/teams/vro-restricted/members) team:
- [9. Delete old Pre-Releases and tags](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/6089889581)
- [9. (Internal) Delete old published non-DEV image](https://github.com/department-of-veterans-affairs/abd-vro-internal/actions/runs/6089925067)
- [9. Delete old workflow runs](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/6089929653)
- [prod: Deploying secrets](https://github.com/department-of-veterans-affairs/abd-vro-internal/actions/runs/6089975270)
- [prod: Update deployment svc-lighthouse-api v3.4.0](https://github.com/department-of-veterans-affairs/abd-vro-internal/actions/runs/6090029005)

Note the `gate-check` job runs for each of the updated workflows:
image

**Comment by chengjie8** [comment]
Great!. I have verified all the workflows in Yoom's comments.

**Review by chengjie8 — APPROVED** [review]
LGTM!

File: .github/workflows/delete-published-images.yml [hunk 1]
```diff
@@ -34,7 +34,32 @@ on:
...
+    runs-on: ubuntu-latest
+    steps:
+    - id: checkUserMember
+      # Only check for manual runs, not scheduled runs
+      if: github.event_name == 'workflow_dispatch'
+      uses: tspascoal/get-user-teams-membership@v2
+      with:
+        username: ${{ github.actor }}
+        team: 'vro-restricted'
+        GITHUB_TOKEN: ${{ secrets.ACCESS_TOKEN_READ_TEAM }}
```
Comment (tejans24) [inline .github/workflows/delete-published-images.yml:48]: Should this be setup as a reusable block?

**Review by tejans24 — APPROVED** [review]
LGTM

File: .github/workflows/delete-published-images.yml (hunk 1 above)
Comment (yoomlam) [reply .github/workflows/delete-published-images.yml:48]: I considered that, but there's 2 different versions. Maybe someone can refactor that later.

**Pull request merged**